base/filesystem.jl

```diff
@@ -4,6 +4,37 @@
 
 module Filesystem
 
+"""
+    JL_O_APPEND
+    JL_O_ASYNC
+    JL_O_CLOEXEC
+    JL_O_CREAT
+    JL_O_DIRECT
+    JL_O_DIRECTORY
+    JL_O_DSYNC
+    JL_O_EXCL
+    JL_O_LARGEFILE
+    JL_O_NOATIME
+    JL_O_NOCTTY
+    JL_O_NONBLOCK
+    JL_O_PATH
+    JL_O_RANDOM
+    JL_O_RDONLY
+    JL_O_RDWR
+    JL_O_RSYNC
+    JL_O_SEQUENTIAL
+    JL_O_SHORT_LIVED
+    JL_O_SYNC
+    JL_O_TEMPORARY
+    JL_O_TMPFILE
+    JL_O_TRUNC
+    JL_O_WRONLY
+
+Enum constant for the `open` syscall, where `JL_O_*` corresponds to the `O_*` constant.
+See your platform man page for details, if applicable.
```

Comment:
The page doesn't explain all of them, so I rather not link it here

Reply:
Which is missing? I am also a maintainer of that documentation, so it would help to know if one is missing from it

Reply:
`ASYNC, CLOEXEC, LARGEFILE, PATH, RSYNC, TMPFILE` are missing from the libuv docs.
And there are some in the libuv docs that are not there in julia.
But if you say that this is the ultimate docu for this, I'll add a link to the docstring. 
```suggestion
See [the libuv docs](https://docs.libuv.org/en/v1.x/fs.html#file-open-constants) for more details.
```

Reply:
We should probably delete `ASYNC`, `CLOEXEC`, `LARGEFILE`, and `RSYNC` exports from our list, since we neither define nor support any of those either. Looks like `O_PATH` and `O_TMPFILE` are the only (linux-specific) ones we might pass through successfully, so we could put an asterisk next to those noting they are linux-only.